### PR TITLE
Feature/node login app bar

### PR DESCRIPTION
**Pull request by mjadach-iv**

### CODE REVIEW

File: src/future-hopr-lib-components/Navbar/navBarItems.tsx [hunk 1]
```diff
@@ -25,8 +34,9 @@ const NavBarItems = (props) => {
     : 'poz-undefined';
 
   return (
-    <Content className={['menu', `menu-${poz}`].join(' ')}>
-      {props.itemsNavbar[0]?.label &&
+    <Content className={['menu', `menu-${poz}`, props.webapp ? 'menu-webpapp' : ''].join(' ')}>
```
Comment (camilovegag) [inline src/future-hopr-lib-components/Navbar/navBarItems.tsx:37]: I think `position` can be clearer than `poz`

File: src/router.tsx [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
 import MailIcon from '@mui/icons-material/Mail';
 import MenuIcon from '@mui/icons-material/Menu';
 import InboxIcon from '@mui/icons-material/MoveToInbox';
-import Layout from './future-hopr-lib-components/Layout';
-import AliasesPage from './sections/aliases';
-import InfoPage from './sections/info';
+
+// Types
+import { Store } from './types/index';
+import { useEffect } from 'react';
```
Comment (camilovegag) [inline src/router.tsx:27]: This shouldn't be under // Types

File: src/sections/selectNode.tsx [hunk 3]
```diff
@@ -20,47 +21,66 @@ function Section1() {
...
-    (store: Store) => store.auth.status.connecting
-  );
-  const connected = useAppSelector(
-    (store: Store) => store.auth.status.connected
-  );
+  const connecting = useAppSelector((store: Store) => store.auth.status.connecting);
+  const connected = useAppSelector((store: Store) => store.auth.status.connected);
+  const loginData = useAppSelector((store: Store) => store.auth.loginData);
+  
+  const [searchParams, setSearchParams] = useSearchParams();
```
Comment (camilovegag) [inline src/sections/selectNode.tsx:35]: should `setSearchParams` be `set_searchParams`? Or we only do that for states

File: src/sections/selectNode.tsx [hunk 4]
```diff
@@ -20,51 +21,70 @@ function Section1() {
...
+  const [localName, set_localName] = useState(loginData.localName ? loginData.localName : '');
+  const [apiEndpoint, set_apiEndpoint] = useState(loginData.apiEndpoint ? loginData.apiEndpoint : '');
+  const [apiToken, set_apiToken] = useState(loginData.apiToken ? loginData.apiToken : '');
+  const [saveApiToken, set_saveApiToken] = useState(loginData.apiToken ? true : false);
+  const [nodesSavedLocallyChosenIndex, set_nodesSavedLocallyChosenIndex] = useState('' as number | '');
+
+  useEffect(()=>{
+    // Update the Select based on loginData from the Store
+    if(!loginData.apiEndpoint) return;
+    const existingItem = nodesSavedLocally.findIndex((item: any) => (item.apiEndpoint === loginData.apiEndpoint));
```
Comment (camilovegag) [inline src/sections/selectNode.tsx:45]: Do we have types for this? so we know what `item` has instead of any

File: src/sections/selectNode.tsx [hunk 5]
```diff
@@ -94,15 +116,24 @@ function Section1() {
       >
         Clear local nodes
       </button>
-      apiEndpoint:
+      <br/>
+      Local name:
```
Comment (camilovegag) [inline src/sections/selectNode.tsx:120]: Would it be good to wrap this in a `<label>` tag?

File: src/utils/functions.ts [hunk 6]
```diff
@@ -7,3 +7,27 @@ export function getObjectFromLocalStorage(key: string) {
     return null;
   }
 }
+
+
+export function bubbleSortObject(arr: any[], key: string | number) {
+  
+  for (var i = 0; i < arr.length; i++) {
```
Comment (camilovegag) [inline src/utils/functions.ts:14]: `let` instead of `var`?

File: src/store/slices/auth/initialState.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { getObjectFromLocalStorage } from '../../../utils/functions';
 
-const ADMIN_UI_NODE_LIST = getObjectFromLocalStorage('admin-ui-node-list');
+let ADMIN_UI_NODE_LIST = getObjectFromLocalStorage('admin-ui-node-list');
```
Comment (camilovegag) [inline src/store/slices/auth/initialState.ts:3]: why `let`?

File: src/components/ConnectNode/index.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            >Logout</button>
+        </div>
+        :
+        <SLink
+          to={'node/connect'}
+        >
+          <div>
+            <button
+              onClick={()=>{navigate('node/connect')}}
+            >Connect to Node</button>
```
[the remark this answers is not in the document]
Comment (mjadach-iv) [reply src/components/ConnectNode/index.tsx:62]: Will be fixed on merge of your PR

File: src/future-hopr-lib-components/Navbar/navBar.tsx [hunk 9]
```diff
@@ -4,21 +4,26 @@ import styled from '@emotion/styled';
...
   border-bottom: 1px lightgray solid;
   box-shadow: unset;
-  padding-left: 16px;
-  padding-right: 16px;
-
-  ${(props) =>
-    props.tallerNavBarOnMobile &&
+  ${ (props) => !props.webapp && `
+    padding-left: 16px;
+    padding-right: 16px;
```
[the remark this answers is not in the document]
Comment (mjadach-iv) [reply src/future-hopr-lib-components/Navbar/navBar.tsx:24]: The PRs are not subject to styling

File: src/future-hopr-lib-components/Layout/drawer.tsx [hunk 10]
```diff
@@ -14,6 +14,9 @@ import ListItemText from '@mui/material/ListItemText';
 import Toolbar from '@mui/material/Toolbar';
 import Typography from '@mui/material/Typography';
 
+// Types
+import { ApplicationMapType } from '../../router';
```
[the remark this answers is not in the document]
Comment (mjadach-iv) [reply src/future-hopr-lib-components/Layout/drawer.tsx:18]: Ok

File: src/future-hopr-lib-components/Navbar/navBarItems.tsx [hunk 11]
```diff
@@ -15,7 +16,15 @@ const Content = styled.div`
...
-const NavBarItems = (props) => {
+const NavBarItems: React.FC<{
+  className?: string;
+  center?: boolean;
+  webapp?: boolean;
+  right?: boolean;
+  mobile?: boolean;
+  itemsNavbar?: any[];
+  onButtonClick?: any;
+}> = (props) => {
```
[the remark this answers is not in the document]
Comment (mjadach-iv) [reply src/future-hopr-lib-components/Navbar/navBarItems.tsx:27]: Ok, please update it

File: src/sections/selectNode.tsx (hunk 3 above)
Comment (mjadach-iv) [reply src/sections/selectNode.tsx:35]: Ok, please update it

File: src/sections/selectNode.tsx (hunk 4 above)
Comment (mjadach-iv) [reply src/sections/selectNode.tsx:45]: We can add them, please do it

**Comment by camilovegag** [comment]
Made the code update, missing `values` fix at selectNode.tsx

File: src/utils/functions.ts [hunk 12]
```diff
@@ -7,3 +7,27 @@ export function getObjectFromLocalStorage(key: string) {
     return null;
   }
 }
+
+
+export function bubbleSortObject(arr: any[], key: string | number) {
```
Comment (ItsLuized) [inline src/utils/functions.ts:12]: Do we currently need an implementation of a sorting algorithm or is it to future proof the utils? If it's the second case, we might want to look at other algorithms for sorting for better efficiency.

**Review by ItsLuized — APPROVED** [review]

File: src/utils/functions.ts (hunk 12 above)
Comment (camilovegag) [reply src/utils/functions.ts:12]: Good question
I'm not sure but I think @mjadach-iv can answer it. In my opinion we should merge this PR if everything is working fine and update the sorting algorithm in the future after the doubts are solved :)

Comment (mjadach-iv) [reply src/utils/functions.ts:12]: In future we will can switch when we will need more performance. Good catch.

**Pull request merged**